.github/workflows/teslight-update-packaging-tool-and-tup-file-build.yml

```diff
@@ -0,0 +1,78 @@
+name: TesLight Update Packaging Tool and TUP file Build
+on:
+  push:
+    branches: ["main", "releases/**", "feat/DevOps_Auto_build_for_Update_Packages"]
+    tags: ["v*", "v**", "v.*", "v.**" ]
+  pull_request:
+    branches: ["main", "releases/**"]
+  workflow_dispatch:
+  workflow_call:
+jobs:
+  build_update_tool_for_different_os:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    # prevent stuck jobs
+    timeout-minutes: 20
```

Comment:
20 minutes seems like a lot? Have you an average value how long it took?

Reply:
The timeout is just an additional layer of robustness for our DevOps, standard is 6h :-)

So windows and macOS runners are really slow, the build on Windows can sometimes take 10–12 minutes, that's why I choose 20 minutes.